File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-7.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            m = (l + r) // 2

            if nums[m] == target: return m
            elif nums[r] < nums[m]: 
                if nums[m] <= target or target <= nums[r]: l = m + 1
                else: r = m - 1
            elif nums[m] < nums[l]:
                if target <= nums[m] or nums[l] <= target: r = m - 1
                else: l = m + 1
            else:
                if nums[m] < target: l = m + 1
                else: r = m - 1
        
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-7.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            int remainder = target - nums[i];

            for (int j = 0; j < nums.length; j++) {
                if (i != j && nums[j] == remainder) {
                    return new int[]{i, j};
                }
            }
        }
        
        return new int[]{-1, -1};
    }
}
